Fix off-by-one in WTF::findIgnoringASCIICaseWithoutLength

jjgriego opened the pull request:
#### d565c2070f7adafe2511904b6a3a27f3d93396d2
<pre>
Fix off-by-one in WTF::findIgnoringASCIICaseWithoutLength
<a href="https://bugs.webkit.org/show_bug.cgi?id=256606">https://bugs.webkit.org/show_bug.cgi?id=256606</a>

Reviewed by Chris Dumez.

A needle identical to the haystack won&apos;t be found by this function because of
the length comparison used. Luckily, this function is unused except in testb3,
so nothing else really needs to be changed.

* Source/WTF/wtf/text/StringCommon.h:
(WTF::findIgnoringASCIICaseWithoutLength):

Canonical link: <a href="https://commits.webkit.org/264088@main">https://commits.webkit.org/264088@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/4ec75e9aa552455a3ff2c87a7fb4705c92606190

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/6660 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/6876 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/7058 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/8251 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/6939 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/6659 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/7962 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/6824 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/9836 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/6778 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/7306 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/6099 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/8338 "Built successfully") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/39/builds/4771 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/6026 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/13855 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/5625 "Built successfully and passed tests") | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/6567 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/6107 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/8802 "Passed tests") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/6248 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/6594 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/5402 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/6797 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/5990 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/1577 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/1573 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/10162 "Built successfully") | | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/24/builds/6980 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/6363 "Built successfully") | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/3/builds/1708 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

pxlcoder commented on Source/WTF/wtf/text/StringCommon.h:
```diff
@@ -417,7 +417,7 @@ inline size_t findIgnoringASCIICaseWithoutLength(const char* source, const char*
     unsigned matchLength = strlen(matchCharacters);
 
-    return matchLength < searchLength ? findIgnoringASCIICase(source, matchCharacters, 0, searchLength, matchLength) : notFound;
+    return matchLength <= searchLength ? findIgnoringASCIICase(source, matchCharacters, 0, searchLength, matchLength) : notFound;
```
Can we add a test for this in `Tools/TestWebKitAPI/Tests/WTF/StringCommon.cpp`?

cdumez reviewed (changes requested):
Needs a test.

cdumez reviewed (approved):
LGTM

jjgriego committed:
Fix off-by-one in WTF::findIgnoringASCIICaseWithoutLength
https://bugs.webkit.org/show_bug.cgi?id=256606

Reviewed by Chris Dumez.

A needle identical to the haystack won't be found by this function because of
the length comparison used. Luckily, this function is unused except in testb3,
so nothing else really needs to be changed.

* Source/WTF/wtf/text/StringCommon.h:
(WTF::findIgnoringASCIICaseWithoutLength):

Canonical link: https://commits.webkit.org/264088@main
